File: java/2653.java
class Solution {
    private int find(int f[], int x) {
        for (int i = 0; i < 50; i++) {
            x -= f[i];
            if (x < 1)
                return i - 50;
        }
        return 0;
    }

    public int[] getSubarrayBeauty(int[] n, int k, int x) {
        int l = n.length;
        int[] f = new int[101], r = new int[l - k + 1];
        for (int i = 0; i < k; i++)
            f[n[i] + 50]++;
        r[0] = find(f, x);
        for (int i = k; i < l; i++) {
            f[n[i - k] + 50]--;
            f[n[i] + 50]++;
            r[i - k + 1] = find(f, x);
        }
        return r;
    }
}

File: python3/2653.py
class Solution:
    def getSubarrayBeauty(self, n: List[int], k: int, x: int) -> List[int]:
        f = [0] * 101

        def find(x):
            for i in range(50):
                x -= f[i]
                if x < 1:
                    return i - 50
            return 0

        r = []
        for i in range(k):
            f[n[i] + 50] += 1
        r.append(find(x))
        for i in range(k, len(n)):
            f[n[i - k] + 50] -= 1
            f[n[i] + 50] += 1
            r.append(find(x))
        return r
